numpydoc/xref.py

```diff
@@ -0,0 +1,144 @@
+import re
+from sphinx import addnodes
+
+# When sphinx (including the napoleon extension) parses the parameters
+# section of a docstring, it converts the information into field lists.
+# Some items in the list are for the parameter type. When the type fields
+# are processed, the text is split and some tokens are turned into
+# pending_xref nodes. These nodes are responsible for creating links.
+#
+# numpydoc does not create field lists, so the type information is
+# not placed into fields that can be processed to make links. Instead,
+# when parsing the type information we identify tokens that are link
+# worthy and wrap them around a special role (xref_param_type_role).
+# When the role is processed, we create pending_xref nodes which are
+# later turned into links.
+
+
+QUALIFIED_NAME_RE = re.compile(
+    # e.g int, numpy.array, ~numpy.array, .class_in_current_module
+    r'^'
+    r'[~\.]?'
+    r'[a-zA-Z_]\w*'
+    r'(?:\.[a-zA-Z_]\w*)*'
+    r'$'
+)
+
+CONTAINER_SPLIT_RE = re.compile(
+    # splits dict(str, int) into
+    #    ['dict', '[', 'str', ', ', 'int', ']', '']
+    r'(\s*[\[\]\(\)\{\},]\s*)'
+)
+
+DOUBLE_QUOTE_SPLIT_RE = re.compile(
+    # splits 'callable ``f(x0, *args)`` or ``f(x0, y0, *args)``' into
+    #    ['callable ', '``f(x0, *args)``', ' or ', '``f(x0, y0, *args)``', '']
+    r'(``.+?``)'
+)
+
+ROLE_SPLIT_RE = re.compile(
+    # splits to preserve ReST roles
+    r'(:\w+:`.+?(?<!\\)`)'
```

Comment:
I haven't checked: is ` allowed in anchor text?

Reply:
Yes. But that regex will not catch pathological use use  of backslashes and ticks.